### PR TITLE
Add Topology Map and remove Service Map widget

### PR DESCRIPTION
**Pull request by estherk15**
### What does this PR do?
- Add topology map to menus.en.yaml file
- Remove the Service Map widget from the menu and the index list
- Add information about the widget in the Topology Map page

### Motivation
[DOCS-4081](https://datadoghq.atlassian.net/browse/DOCS-4081)

### Additional Notes
Will remove the Service Map page in a separate [PR](https://github.com/DataDog/documentation/pull/15769)

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by kayayarai** [comment]
Opened editorial review https://datadoghq.atlassian.net/browse/DOCS-4315

File: content/en/dashboards/widgets/topology_map.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+The dedicated [widget JSON schema definition][3] for the service map widget is:
+
+{{< dashboards-widgets-api >}}
+
+## Further Reading
+
+{{< partial name="whats-next/whats-next.html" >}}
+
+[1]: /tracing/services/services_map/
+[2]: /api/v1/dashboards/
```
Comment (hestonhoffman) [inline content/en/dashboards/widgets/topology_map.md:41]:
```suggestion
[2]: /api/latest/dashboards/
```
This link seems to be 404ing, even though it redirects when I try it on production 🤔 
Seeing as though it redirects to latest anyway, let's try that and see if it fixes it.

File: content/en/dashboards/widgets/topology_map.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+---
+title: Topology Map Widget
+kind: documentation
+description: "Displays a map of a service to all of the services that call it, and all of the services that it calls."
+widget_type: "topologymap"
```
Comment (hestonhoffman) [inline content/en/dashboards/widgets/topology_map.md:5]:
```suggestion
widget_type: "topology_map"
```
This will fix the Model and Example sections.

**Review by hestonhoffman — COMMENTED** [review]
I made some edits to fix the output. I didn't know about that widget_type field, so TIL!

**Review by hestonhoffman — APPROVED** [review]

**Pull request merged**